### PR TITLE
Add diverse deid

### PR DESCRIPTION
**Pull request by cascadianblue**
Allow callers of the de-identifer server to specify arbitrary combinations of annotation types and de-identification strategies.

### CODE REVIEW

File: openapi/phi-deidentifier/paths/deidentified-note.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  summary: Get deidentified note
+  description: Returns the deidentified note
+  operationId: "deidentified_note.read"
+  requestBody:
+    content:
+      application/json:
+        schema:
+          $ref: ../components/schemas/DeidentifyRequest.yaml
+  responses:
+      '200':
```
Comment (tschaffter) [inline openapi/phi-deidentifier/paths/deidentified-note.yaml:13]: POST should return the status 201 (already corrected in some of the other API)

File: openapi/phi-deidentifier/components/responses/DeidentifyResponse.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+type: object
+description: response to deidentification request, including annotations with indexes in both the original note and the
```
Comment (tschaffter) [inline openapi/phi-deidentifier/components/responses/DeidentifyResponse.yaml:2]: - Start `description` with a capital letter
- Try to keep it as concise as possible

File: openapi/phi-deidentifier/components/responses/DeidentifyResponse.yaml [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+  deidentified note.
+properties:
+  originalAnnotations:
+    $ref: ../../../commons/components/schemas/Annotation.yaml
```
Comment (tschaffter) [inline openapi/phi-deidentifier/components/responses/DeidentifyResponse.yaml:6]: It looks like we can't add a description to properties that use `$ref`, see https://github.com/OAI/OpenAPI-Specification/issues/1514

File: openapi/phi-deidentifier/components/schemas/DeidentificationConfig.yaml [hunk 4]
```diff
@@ -0,0 +1,31 @@
+type: object
```
Comment (tschaffter) [inline openapi/phi-deidentifier/components/schemas/DeidentificationConfig.yaml:1]: - Add object description

File: openapi/phi-deidentifier/components/schemas/DeidentificationConfig.yaml [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  deidentificationStrategy:
+    type: object
+    description: the method/strategy used to de-identify the note's text
+    properties:
+      maskingCharConfig:
+        type: object
+        properties:
+          maskingChar:
+            type: string
+            maxLength: 1
```
Comment (tschaffter) [inline openapi/phi-deidentifier/components/schemas/DeidentificationConfig.yaml:12]: - Add `minLength: 1`

File: openapi/phi-deidentifier/components/schemas/DeidentificationConfig.yaml [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+        properties:
+          offsetDays:
+            type: integer
+  annotationTypes:
+    type: array
+    description: the types of annotations to which the de-identifer should apply the selected strategy
+    items:
+      type: string
+      enum:
+        - "STREET_ADDRESS"
```
Comment (tschaffter) [inline openapi/phi-deidentifier/components/schemas/DeidentificationConfig.yaml:29]: Use:
- `text_date`
- `text_person_name`
- `text_physical_address`

These names are consistent with the properties of the Annotation.yaml object. We use lower snake case for enum.

File: openapi/phi-deidentifier/components/schemas/DeidentifyRequest.yaml [hunk 7]
```diff
@@ -0,0 +1,14 @@
+type: object
+description: A request to de-identify a note
+properties:
+  deidentifications:
```
Comment (tschaffter) [inline openapi/phi-deidentifier/components/schemas/DeidentifyRequest.yaml:4]: What about renaming to be more descriptive, e.g. `deidentificationStrategies` or `deidentificationConfigurations`?

The selected property name should be the plural version of the schema name used for the `items`.

File: openapi/phi-deidentifier/paths/deidentified-note.yaml [hunk 8]
```diff
@@ -0,0 +1,20 @@
+post:
```
Comment (tschaffter) [inline openapi/phi-deidentifier/paths/deidentified-note.yaml:1]: Rename file to `deidentifiedNote.yaml`

File: openapi/phi-deidentifier/paths/deidentified-note.yaml [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+  tags:
+    - Deidentified Notes
+  summary: Get deidentified note
+  description: Returns the deidentified note
+  operationId: "deidentified_note.read"
```
Comment (tschaffter) [inline openapi/phi-deidentifier/paths/deidentified-note.yaml:6]: Rename operationId to `deidentifiedNote.read`

**Review by tschaffter — CHANGES_REQUESTED** [review]

File: openapi/phi-deidentifier/components/responses/DeidentifyResponse.yaml (hunk 3 above)
Comment (cascadianblue) [reply openapi/phi-deidentifier/components/responses/DeidentifyResponse.yaml:5]: Is there some way I can clarify what these properties represent, other than through their field names?

**Comment by cascadianblue** [comment]
There was one linter warning I wasn't sure how to get rid of:

```
  Message :   operationIds must follow case convention: lower_camel_case
  Path    :   paths./deidentifiedNote.post.operationId
  Line    :   33
```

Any idea how to deal with that?

**Pull request merged**